src/monitor/commands/edit-monitor-availability.command.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/database/prisma.service';
+import { days } from '../dto/monitor-availability.dto';
+import {
+  EditMonitorBodyMissingFieldsException,
+  MonitoringNotFoundException,
+  UserLoggedNotResponsableForMonitoringException,
+} from '../utils/exceptions';
+import { EditMonitorAvailabilityDTO } from '../dto/edit-monitor-availability.dto';
+
+@Injectable()
+export class EditMonitorAvailabilityCommand {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async execute(
+    monitorId: number,
+    tokenUserId: number,
+    preferentialPlace: string,
+    availability: days[],
+  ): Promise<EditMonitorAvailabilityDTO> {
+    if (!preferentialPlace && !availability)
+      throw new EditMonitorBodyMissingFieldsException();
+
+    const monitoring = await this.prisma.monitor.findFirst({
+      where: {
+        id: monitorId,
+      },
+    });
+
+    if (!monitoring) throw new MonitoringNotFoundException();
+
+    if (monitoring.student_id != tokenUserId)
+      throw new UserLoggedNotResponsableForMonitoringException();
+
+    await this.prisma.$transaction(async (tx) => {
+      let monitorSettings;
+      let availableTimes;
+
+      if (preferentialPlace) {
+        await tx.monitorSettings.updateMany({
+          where: { monitor_id: monitorId, is_active: true },
+          data: {
+            is_active: false,
+          },
+        });
+
+        monitorSettings = await tx.monitorSettings.create({
+          data: {
+            monitor_id: monitorId,
+            preferential_place: preferentialPlace,
+            is_active: true,
+          },
+        });
+      }
+
+      if (availability) {
+        await tx.availableTimes.deleteMany({
+          where: { monitor_id: monitorId },
+        });
+
+        const formattedAvailableTimes: any[] = [];
+
+        availability.forEach((day) =>
+          day.hours.forEach((hour) =>
+            formattedAvailableTimes.push({
+              monitor_id: monitorId,
+              week_day: day.weekDay,
+              start: hour.start,
+              end: hour.end,
+            }),
+          ),
+        );
+
+        availableTimes = await tx.availableTimes.createMany({
+          data: formattedAvailableTimes,
+        });
+      }
+
+      if (!monitorSettings) {
+        monitorSettings = await tx.monitorSettings.findFirst({
+          where: { monitor_id: monitorId, is_active: true },
+        });
+      }
+
+      if (!availableTimes) {
+        availableTimes = await tx.availableTimes.findMany({
+          where: { monitor_id: monitorId },
+        });
+      }
+
+      if (monitorSettings && availableTimes) {
+        await tx.monitor.update({
+          where: { id: monitorId },
+          data: {
+            id_status: 3,
+          },
+        });
+      }
+    });
+
+    const response: EditMonitorAvailabilityDTO = {
+      preferentialPlace: preferentialPlace,
+      availability: availability,
```

Comment:
Quando a variável tem o mesmo nome que o campo, não precisa repetir:

```suggestion
      preferentialPlace,
      availability,
```

Reply:
Resolvido